sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ReportPayload.java

```diff
@@ -0,0 +1,143 @@
+        return operation;
+    }
+
+    public void setOperation(OperationType operation) {
+        this.operation = operation;
+    }
+
+    public ResourceType getResource() {
+        return resource;
+    }
+
+    public void setResource(ResourceType resource) {
+        this.resource = resource;
+    }
+
+    public Integer getStatusCode() {
+        return statusCode;
+    }
+
+    public void setStatusCode(Integer statusCode) {
+        this.statusCode = statusCode;
+    }
+
+    public String getOperationId() {
+        return operationId;
+    }
+
+    public void setOperationId(String operationId) {
+        this.operationId = operationId;
+    }
+
+    public MetricInfo getMetricInfo() {
+        return metricInfo;
+    }
+
+    public void setMetricInfo(MetricInfo metricInfo) {
+        this.metricInfo = metricInfo;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        ReportPayload reportPayload = as(obj, ReportPayload.class);
+        if (reportPayload == null) {
+            return false;
+        }
+        if (((reportPayload.regionsContacted == null && this.regionsContacted == null) || (reportPayload.regionsContacted != null && reportPayload.regionsContacted.equals(this.regionsContacted))) &&
+            ((reportPayload.greaterThan1Kb == null && this.greaterThan1Kb == null) || (reportPayload.greaterThan1Kb != null && reportPayload.greaterThan1Kb.equals(this.greaterThan1Kb))) &&
+            ((reportPayload.consistency == null && this.consistency == null) || (reportPayload.consistency != null && reportPayload.consistency.equals(this.consistency))) &&
+            ((reportPayload.databasesName == null && this.databasesName == null) || (reportPayload.databasesName != null && reportPayload.databasesName.equals(this.databasesName))) &&
+            ((reportPayload.containerName == null && this.containerName == null) || (reportPayload.containerName != null && reportPayload.containerName.equals(this.containerName))) &&
+            ((reportPayload.operation == null && this.operation == null) || (reportPayload.operation != null && reportPayload.operation.equals(this.operation))) &&
+            ((reportPayload.resource == null && this.resource == null) || (reportPayload.resource != null && reportPayload.resource.equals(this.resource))) &&
+            ((reportPayload.statusCode == null && this.statusCode == null) || (reportPayload.statusCode != null && reportPayload.statusCode.equals(this.statusCode))) &&
+            ((reportPayload.metricInfo.getMetricsName() == null && this.metricInfo.getMetricsName() == null) || (reportPayload.metricInfo.getMetricsName() != null && reportPayload.metricInfo.getMetricsName().equals(this.metricInfo.getMetricsName())))) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 0;
+        hash = (hash * 397) ^ (this.regionsContacted == null ? 0 : this.regionsContacted.hashCode());
```

Comment:
I got this hashing from our existing partition key class, I could have done without (hash * 397) , thoughts ?

Reply:
Oh I see, intellij with hash 397 generated 1474114674 and without it 1390251919. As you rightly mentioned it will help in better distribution of hash. So we are good with this.